### DISCUSSION TITLE
I'm trying to fetch a .wasm file and it's always 404 no matter where I place it.

### DISCUSSION THREAD

**Discussion by Tyler2k** [preactjs/preact-cli#1610] [Q&A] [answered]
I've placed inside a public folder, I've placed it inside the build folder. No matter where, when I load the app, the file is 404. I'm using "dev": "preact watch" to run the server.

**Comment by rschristian** [comment, marked as answer]
`build/` has no bearing on the dev server. That's your built output, used by the `serve` command in the templates or for upload to a server.

What do you mean by "a public folder"? 

By default, `src/assets/` is copied into your output and you can put static assets there. Or, you can add another path to copy to your build output using `CopyWebpackPlugin`.

**Reply by Tyler2k** [reply]
Thanks for the quick response! I'm trying to get this running on local. So I've got two folders, an assembly script folder that generates the .wasm. And another preact folder. I've built the same project with react instead of preact and it worked fine when I copy pasted the optimized.wasm into the react public folder. But doing the same in preact, it just doesn't see the file. I just tried putting it in the assets folder and running npm run dev again and no luck 404.
![image](https://user-images.githubusercontent.com/17047905/141602826-8a2e0c8c-fd59-4658-83ae-55c87186e746.png)

![image](https://user-images.githubusercontent.com/17047905/141602761-95d0701f-25c8-43db-8bdb-1024b8250d8d.png)

**Reply by Tyler2k** [reply]
![image](https://user-images.githubusercontent.com/17047905/141602899-1cac296d-48fb-420d-b5ae-1808876d9c4c.png)

**Reply by rschristian** [reply]
React vs Preact has no bearing here. They're just UI libraries.

Create-React-App, which is what I assume you mean, copies assets from `public/`. Preact-CLI does not. We use `src/assets/`.

You're using the wrong URL. You need `/assets/optimized.wasm`, as that's where the resource is sat.

Easy way to check this is build your app (`preact build`). Think of `build/` as `/`. You need to request your resources accordingly.

Again, you have full control and can edit this if you'd like using `CopyWebpackPlugin`. Can copy any asset to any output path that you'd like. This is just the default that's supported right out-of-the-box.

**Reply by Tyler2k** [reply]
You are a god damn genius, it worked! I've been pulling my hair out all day. You know when you get tunnel vision on a problem and you just stop thinking clearly and you just seem to get dumber and dumber the longer you try to fix it. Thank you!

**Reply by rschristian** [reply]
No problem. Certainly do ask if you ever get stuck on something, we've definitely all been there many times

[We do also have a slack](https://chat.preactjs.com) if you prefer that format, though nothing wrong with asking here (and this here could help future users).

This is probably a common enough thing where we should provide some documentation.
